### PR TITLE
[feat] fixed auth context + sessionDemo

### PR DESCRIPTION
**Pull request by RohinJ444**
[//]: # "Feel free to customize this template to your project's vibes"

### :croissant: Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

Implemented auth context that saves login / session state and allows for sign in / sign out functionality. Currently, after login or signup, the new code will route to a dummy sessionDemo page that displays which user is currently logged in and provides a button for current user to log out with. I write to the console logs every time a new session is created which could aid in debugging / validating, but I believe I fixed the error of improperly saving the login state we discussed during worksession.

ALSO UPDATE: Refactored signin and sign up code into auth.ts, add users to volunteers table on successful sign up, check if user is in volunteers table at sign up and sign in. 

### :baguette_bread: Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## :sandwich: How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

So the authContextProvider which contains the core functionality of the auth context is located in utils/authContextProvider.tsx

Also made adjustments to app/layout.tsx, app/auth/signin/page.tsx, app/auth/signup/page.tsx, and created a new app/auth/sessionDemo/page.tsx to illustrate that the session management works. 

## :pie: Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Need to adjust routing from dummy sessionDemo page to desired next page, and integrate authContext into all pages

### :pretzel: Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

https://www.youtube.com/watch?v=r7SAlIlMs1k 
https://github.com/calblueprint/girls-write-now/pull/17/files

### :bagel: Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @celinechoiii @me-liu

### CODE REVIEW

File: utils/AuthProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+export function AuthContextProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [session, setSession] = useState<Session | null>(null);
+
+  /* useEffect(() => {
```
Comment (celinechoiii) [inline utils/AuthProvider.tsx]: can you delete the comments?

File: app/(auth)/signin/page.tsx [hunk 2]
```diff
@@ -29,18 +33,35 @@ export default function SignIn() {
...
+      try {
+        const { error } = await sessionHandler.signInWithEmail(email, password);
+        if (error) {
+          setMessage(`Login failed: ${error.message}`);
+          setIsError(true);
+        } else {
+          setMessage('Login successful!');
+          setIsError(false);
+          setTimeout(() => {
+            router.push('/sessionDemo');
```
Comment (celinechoiii) [inline app/(auth)/signin/page.tsx]: can you push `onboarding/general` to the router instead of the sessionDemo?

File: app/(auth)/sessionDemo/page.tsx
Comment (celinechoiii) [inline app/(auth)/sessionDemo/page.tsx:1]: thanks for the including this file for testing! can you unadd it so that i can merge it to main?

Comment (RohinJ444) [reply app/(auth)/sessionDemo/page.tsx:1]: oop yeah
